tests/MongoDB.Driver.Legacy.Tests/Builders/IndexOptionsBuilderTests.cs

```diff
@@ -165,5 +166,17 @@ public void TestMultipleWeights()
             string expected = "{ \"weights\" : { \"a\" : 2, \"b\" : 10 } }";
             Assert.Equal(expected, options.ToJson());
         }
+
+        [Fact]
+        public void WildcardProjection_should_return_expected_result()
+        {
+            var options = IndexOptions.SetWildcardProjection("a", true);
+            BsonDocument expected = BsonDocument.Parse("{ \"wildcardProjection\" : { \"a\" : 1 } }");
+            options.ToBsonDocument().Should().Be(expected);
+
+            options = IndexOptions.SetWildcardProjection("a", true).SetWildcardProjection("b.c", false);
+            expected = BsonDocument.Parse("{ \"wildcardProjection\" : { \"a\" : 1, \"b.c\" :  0} }");
```

Comment:
Presumably we're relying on the server to error out?

Reply:
This test doesn't call a SERVER. Do you mean that the generated wildProjection is unsupported by SERVER?

Reply:
Yes, sorry for the confusion: just checking that the driver isn't validating projections and we're relying on the server to error out when the projection is invalid :)

Reply:
yes, DRIVER does not validate the projection and we expect that the SERVER will throw an exception is it is necessary

Reply:
Thanks for the clarification Dima :)

Reply:
Do we want an example in the tests that is invalid?

Maybe use true for both "a" and "b"?

Reply:
how about using `false` for '_id' field?
Done.